cloudsync/sync/manager.py

```diff
@@ -711,8 +719,7 @@ def create_synced(self, changed, sync, translated_path):  # pylint: disable=too-
 
     def handle_cloud_file_not_found_error(self, changed, sync, synced):
         if sync.priority > 5:
-            log.exception("punted too many times on CloudFileNotFoundError, giving up")
-            return FINISHED
+            raise ex.CloudTooManyRetriesError("punted too many times on CloudFileNotFoundError, giving up")
```

Comment:
Also just for clarity, what's the motivation for adding this new exception type? How else do you envision this being used?

Reply:
The REALLY_FINISHED solution seemed messy and error prone, as it forces us to add more `if` checks throughout. I think we already have a couple of other places where this exception would be useful (didn't make those changes yet) and there could be more in the future.

Also, too many retries does feel like an exception.

Reply:
ah okay right right, I agree with this